PowerPointLabs/Test/FunctionalTest/SyncLabTest.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Drawing;
+using System.Windows.Forms;
+using TestInterface;
+using Microsoft.Office.Interop.PowerPoint;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.Util;
+using Point = System.Drawing.Point;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using PowerPointLabs;
+
+namespace Test.FunctionalTest
+{
+    [TestClass]
+    public class SyncLabTest : BaseFunctionalTest
+    {
+        private const string UnrotatedRectangle = "Rectangle 3";
+        private const string Oval = "Oval 4";
+        private const string RotatedArrow = "Right Arrow 5";
+        private const string CopyFromShape = "CopyFrom";
+
+
+        protected override string GetTestingSlideName()
+        {
+            return "SyncLab.pptx";
+        }
+
+        [TestMethod]
+        [TestCategory("FT")]
+        public void FT_SyncLabTest()
+        {
+            var syncLab = PplFeatures.SyncLab;
+            syncLab.OpenPane();
+
+            TestCopyDialog(syncLab);
+            TestPasteDialog(syncLab);
+        }
+
+        private void TestCopyDialog(ISyncLabController syncLab)
+        {
+            var actualSlide = PpOperations.SelectSlide(4);
+
+            // no selection
+            MessageBoxUtil.ExpectMessageBoxWillPopUp("", "Please select one item to copy.", syncLab.Copy, "Ok");
+
+            // 2 item selected
+            List<String> shapes = new List<string> { Oval, RotatedArrow };
+            PpOperations.SelectShapes(shapes);
+            MessageBoxUtil.ExpectMessageBoxWillPopUp("", "Please select one item to copy.", syncLab.Copy, "Ok");
+
+            // copy successful
+            PpOperations.SelectShape(CopyFromShape);
+            new Task(() =>
+            {
+                ThreadUtil.WaitFor(1000);
+                SendKeys.SendWait("{ENTER}");
+            }).Start();
+            syncLab.Copy();
```

Comment:
Hm.. I tried that and previously and encountered 2 problems

1. The window blocks all execution.
2. The FunctionalTestExtensions class, which is used to get the "_pane" does not work for the pop up. I suspect it is because FunctionalTestExtensions is only able to retrieve a Windows Form object. Does this mean I will have to wrap the pop up in a Windows Form? Is there anyway to just retrieve a WPF object?